### ISSUE TITLE
I2C writeRead method compares written to writeOffset instead of writeSize

### ISSUE THREAD

**Issue by kasperski12** [Pi4J/pi4j#388]
Hi all
I think that there is bug in default implementation of writeRead method in I2c.java
```
default int writeRead(byte[] writeBuffer, int writeSize, int writeOffset, byte[] readBuffer, int readSize,
        int readOffset) {
        return execute(() -> {
            int written = write(writeBuffer, writeOffset, writeSize);
            if (written != writeOffset)
                throw new IllegalStateException(
                    "Expected to write " + writeOffset + " bytes but only wrote " + written + " bytes");
            return read(readBuffer, readOffset, readSize);
        });
    }
```
The if conditions compares written to writeOffset instead of writeSize which is the number of bytes to write.

**Comment by eitch** [comment]
Thanks, good catch! There really was wrong! It will be fixed in the next release!

**Comment by FDelporte** [comment]
Included in 2.7.0
